0038-http-auth-profiles/http-auth-profiles.md

```diff
@@ -0,0 +1,191 @@
+
+* **Cons**
+  * More complex than the `SIMPLE` and `JWT_HS_256` profile.
+  * Total transmitted bytes for authentication are more than the `SIMPLE` and `JWT_HS_256` schemes. However, HTTP/2 header compression should mitigate this differential as well.
+  * Requires out-of-band communication for both peers to agree upon public keys.
+  * Verification performance is slower than the `SIMPLE` and `JWT_HS_256` profile`.
+
+### Authorization
+The `HS-256` and `RS_256` profiles defined in this RFC rely upon signed JWTs, which support arbitrary claims that can be used for authorization decisions. This document does not define any authorization-specific primitives, although implementations MAY use various authentication claims in order to inform authorization decisions.
+
+## Appendix1: Security Best Practices
+This section outlines and clarifies some best practices for authentication-token security when using this protocol. Recommendations in this section are _Non-Normative_, but highly RECOMMENDED.
+
+### Follow Standardized Security Recommendations
+It is advisable to follow all applicable best practices when using a Bearer-token scheme for authentication. [Section 5.1 of RFC-6570](https://tools.ietf.org/html/rfc6750#section-5) contains a number of good practices that should be considered on a per-deployment basis. 
+
+### Use SIMPLE Profile for Development/Testing Only
+The `SIMPLE` authentication profile provides only marginal benefits when compared to the `JWT_HS_256` profile, but introduces significant drawbacks as outlined in the "Trade-off Summary" sections of this RFC. As such, the `SIMPLE` profile MAY be used for development or testing purposes, but SHOULD NOT be used in production scenarios. Instead, prefer `JWT_HS_256` for production deployments.
+
+### Avoid HTTP Basic and Form-based Auth
+HTTP Auth schemes using a username and password are NOT RECOMMENDED for the same reasons that the `SIMPLE` profile is only recommended for development and testing scenarios.
+ 
+### Use Reasonable Token Expiries
+Tokens generators should choose a reasonable token expiry. Considerations in this choice include the ability to cache and re-use tokens for a limited time in order to enable very fast authentication decisions. However, this should be balanced against a desire for shorter token lifetimes, which will limit the attack surface caused by a compromised token.
+
+As a best practice, implementations SHOULD use tokens that expire. For example, consider generating tokens with a lifetime that doesn't exceed 5 minutes.
+
+### Secrets At Rest
+Implementations SHOULD protect secret-values that can be used to generate authentication tokens by encrypting them prior to storage. This will help prevent actual shared-secrets or other sensitive data from being captured by unauthorized parties, increasing the chances that only Interledger software will be able to generate tokens.
```

Comment:
This seems like it's worth defining as a `MUST` to me.

Reply:
Taking an opportunity to expand on this since I was perhaps being a bit vague.

Since encrypting data at rest implies that it require some key to be able to decrypt and make the data usable, you can apply an encryption key either on the data itself before being stored on a disk, or you can have the partition be encrypted itself (which would also require a key to mount and decrypt).

Since the verbiage above says `prior to storage` it made me think that the latter condition should be considered as well since it could also be used as a means of encryption at rest, but wouldn't be predicated on `prior` since you're letting the storage encrypt for you.

Reply:
Ah, I see now. How about this change in verbiage:

```text
Implementations SHOULD protect secret-values that can be used to generate authentication tokens by encrypting them prior to storage, and/or storing them in an encrypted storage device.
```

Reply:
Just pushed this change. Feel free to suggest something else if you don't think the new text is sufficient.

Reply:
Yes; I like that verbiage. My sticking point is still `SHOULD` vs `MUST` I guess. Part of me thinks we should be more forceful since compromising this information is a big deal.